crates/sequencer/src/sequencer.rs

```diff
@@ -941,12 +941,14 @@ where
                             // This is mainly to make sure we account for the execution time to
                             // achieve consistent 2-second block production.
                             let parent_block_exec_time = instant.elapsed();
+                            let mut next_block_production_duration = target_block_time
```

Comment:
Why do we even need this one?

> The difference between interval and [sleep](https://docs.rs/tokio/latest/tokio/time/fn.sleep.html) is that an [Interval](https://docs.rs/tokio/latest/tokio/time/struct.Interval.html) measures the time since the last tick, which means that [.tick().await](https://docs.rs/tokio/latest/tokio/time/struct.Interval.html#method.tick) may wait for a shorter time than the duration specified for the interval if some time has passed between calls to [.tick().await](https://docs.rs/tokio/latest/tokio/time/struct.Interval.html#method.tick).

So creating an Interval(2sec) is not enough?